package/MDAnalysis/coordinates/base.py

```diff
@@ -1589,6 +1608,24 @@ class Writer(IObase):
     See Trajectory API definition in :mod:`MDAnalysis.coordinates.__init__` for
     the required attributes and methods.
     """
+    class __metaclass__(type):
+        # Auto register upon class creation
+        def __init__(cls, name, bases, classdict):
+            type.__init__(type, name, bases, classdict)
```

Comment:
This would break if you somehow subclassed `__metaclass__` elsewhere in the class... So I think techinically I should user super, but it's harmless not to?